bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java

```diff
@@ -0,0 +1,247 @@
+        } else {
+            if (maxMediaVolume == 0)
+                return; // We can not transform percentage
+            int targetVolume = Integer.parseInt(command.toFullString());
+            adbConnection.setMediaVolume((int) Math.round(fromPercent(targetVolume, maxMediaVolume)));
+            updateState(channelUID, new PercentType(targetVolume));
+        }
+    }
+
+    private double toPercent(double value, double maxValue) {
+        return (value / maxValue) * 100;
+    }
+
+    private double fromPercent(double value, double maxValue) {
+        return (value / 100) * maxValue;
+    }
+
+    private void handleMediaControlCommand(ChannelUID channelUID, Command command) throws InterruptedException,
+            IOException, AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceNotConnectedException {
+        if (command instanceof RefreshType) {
+            boolean playing = adbConnection.isPlayingMedia();
+            updateState(channelUID, playing ? PlayPauseType.PLAY : PlayPauseType.PAUSE);
+        } else if (command instanceof PlayPauseType) {
+            if (command == PlayPauseType.PLAY) {
+                adbConnection.sendKeyEvent(KEY_EVENT_PLAY);
+                updateState(channelUID, PlayPauseType.PLAY);
+            } else if (command == PlayPauseType.PAUSE) {
+                adbConnection.sendKeyEvent(KEY_EVENT_PAUSE);
+                updateState(channelUID, PlayPauseType.PAUSE);
+            }
+        } else if (command instanceof NextPreviousType) {
+            if (command == NextPreviousType.NEXT) {
+                adbConnection.sendKeyEvent(KEY_EVENT_NEXT);
+            } else if (command == NextPreviousType.PREVIOUS) {
+                adbConnection.sendKeyEvent(KEY_EVENT_PREVIOUS);
+            }
+        } else if (command instanceof RewindFastforwardType) {
+            if (command == RewindFastforwardType.FASTFORWARD) {
+                adbConnection.sendKeyEvent(KEY_EVENT_MEDIA_FAST_FORWARD);
+            } else if (command == RewindFastforwardType.REWIND) {
+                adbConnection.sendKeyEvent(KEY_EVENT_MEDIA_REWIND);
+            }
+        } else {
+            logger.warn("Unknown media control command: {}", command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        var currentConfig = getConfigAs(AndroidDebugBridgeConfiguration.class);
+        config = currentConfig;
+        adbConnection.configure(currentConfig.ip, currentConfig.port);
+        updateStatus(ThingStatus.UNKNOWN);
+        connectionCheckerSchedule = scheduler.scheduleWithFixedDelay(this::checkConnection, 0,
+                currentConfig.refreshTimeSec, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        var schedule = connectionCheckerSchedule;
+        if (schedule != null) {
+            schedule.cancel(true);
+            connectionCheckerSchedule = null;
+        }
+        adbConnection.disconnect();
```

Comment:
How long can this take at maximum? `dispose()` is supposed to return fast.

Reply:
I made some test and it takes less than 5 ms always.

Reply:
I've never face that problem when I was testing the device disconnection and reconnection. Also I have reviewed the implementation and I don't see any reason why its execution could be blocked by a network problem.